File: lib/data/models/promotion.dart
enum PromotionMediaType { image, video }

class Promotion {
  const Promotion({
    required this.id,
    required this.shopId,
    required this.shopName,
    required this.headline,
    required this.body,
    required this.mediaType,
    this.mediaUrl,
    this.thumbUrl,
    this.badge,
    this.ctaLabel = 'Order now',
    required this.createdAt,
  });

  final String id;
  final String shopId;
  final String shopName;
  final String headline;
  final String body;
  final PromotionMediaType mediaType;
  final String? mediaUrl;
  final String? thumbUrl;
  final String? badge;
  final String ctaLabel;
  final DateTime createdAt;
}

File: backend/src/models/Promotion.js
const mongoose = require('mongoose');

const promotionSchema = new mongoose.Schema(
  {
    promotionId: { type: String, required: true, unique: true },
    shopId: { type: String, required: true },
    headline: { type: String, required: true },
    body: { type: String, default: '' },
    mediaUrl: { type: String, default: null },
    mediaType: {
      type: String,
      enum: ['image', 'video', 'none'],
      default: 'image',
    },
    badge: { type: String, default: null },
    ctaLabel: { type: String, default: null },
    isActive: { type: Boolean, default: true },
    startsAt: { type: Date, default: null },
    endsAt: { type: Date, default: null },
  },
  { timestamps: true },
);

promotionSchema.index({ shopId: 1, isActive: 1 });

module.exports = mongoose.model('Promotion', promotionSchema);
